### PR TITLE
Add changes to allow use of robot model

### PR DESCRIPTION
**Pull request by evelyd**
Small changes to make it possible for the user to specify the joint position for calibration (to allow robot to stand in T-pose), and to allow the floor contact to be offset by the height of the foot frame being used.

### CODE REVIEW

File: src/IK/include/BiomechanicalAnalysis/IK/InverseKinematics.h [hunk 1]
```diff
@@ -449,7 +449,7 @@ class HumanIK
      * @param frameRef reference frame used as world
      * @return true if the calibration matrix is set correctly
      */
-    bool calibrateAllWithWorld(std::unordered_map<int, nodeData> nodeStruct, std::string frameRef = "");
+    bool calibrateAllWithWorld(std::unordered_map<int, nodeData> nodeStruct, Eigen::VectorXd jointPositions, std::string frameRef = "");
```
Comment (GiulioRomualdi) [inline src/IK/include/BiomechanicalAnalysis/IK/InverseKinematics.h:452]:
```suggestion
    bool calibrateAllWithWorld(std::unordered_map<int, nodeData> nodeStruct, Eigen::Ref<const Eigen::VectorXd> jointPositions, std::string frameRef = "");
```

File: src/IK/include/BiomechanicalAnalysis/IK/InverseKinematics.h [hunk 2]
```diff
@@ -439,7 +439,7 @@ class HumanIK
      * @return true if the calibration matrix is set correctly
      * @note gravity is expected to be aligned with the z-axis of the IMU frame
      */
-    bool calibrateWorldYaw(std::unordered_map<int, nodeData> nodeStruct);
+    bool calibrateWorldYaw(std::unordered_map<int, nodeData> nodeStruct, Eigen::VectorXd jointPositions);
```
Comment (GiulioRomualdi) [inline src/IK/include/BiomechanicalAnalysis/IK/InverseKinematics.h:442]:
```suggestion
    bool calibrateWorldYaw(std::unordered_map<int, nodeData> nodeStruct, Eigen::Ref<const Eigen::VectorXd> jointPositions);
```

File: src/examples/IK/exampleIK.cpp [hunk 3]
```diff
@@ -483,7 +491,7 @@ int main()
             getNodeVerticalForce(ifeel_data, node, ii, force);
 
             // IK solver: Update floor contact task for the current node
-            ik.updateFloorContactTask(node, force);
+            ik.updateFloorContactTask(node, force, 0.0);
```
Comment (GiulioRomualdi) [inline src/examples/IK/exampleIK.cpp:494]:
```suggestion
            const double linkHeight = 0.0;
            ik.updateFloorContactTask(node, force, linkHeight);
```

**Review by davidegorbani — CHANGES_REQUESTED** [review]

File: src/IK/include/BiomechanicalAnalysis/IK/InverseKinematics.h [hunk 4]
```diff
@@ -439,7 +439,7 @@ class HumanIK
      * @return true if the calibration matrix is set correctly
      * @note gravity is expected to be aligned with the z-axis of the IMU frame
      */
-    bool calibrateWorldYaw(std::unordered_map<int, nodeData> nodeStruct);
+    bool calibrateWorldYaw(std::unordered_map<int, nodeData> nodeStruct, Eigen::Ref<const Eigen::VectorXd> jointPositions);
```
[the remark this answers is not in the document]
Comment (evelyd) [reply src/IK/include/BiomechanicalAnalysis/IK/InverseKinematics.h]: Resolved in https://github.com/ami-iit/biomechanical-analysis-framework/pull/51/commits/da4e6635eb43c709a5db812c94bb16003242d969.

File: src/IK/include/BiomechanicalAnalysis/IK/InverseKinematics.h [hunk 5]
```diff
@@ -423,7 +423,7 @@ class HumanIK
      * @param footInContact unordered map containing the node number and the vertical force
      * @return true if the calibration matrix is set correctly
      */
-    bool updateFloorContactTasks(const std::unordered_map<int, Eigen::Matrix<double, 6, 1>>& wrenchMap);
+    bool updateFloorContactTasks(const std::unordered_map<int, Eigen::Matrix<double, 6, 1>>& wrenchMap, const double linkHeight);
```
[the remark this answers is not in the document]
Comment (evelyd) [reply src/IK/include/BiomechanicalAnalysis/IK/InverseKinematics.h]: Resolved in https://github.com/ami-iit/biomechanical-analysis-framework/pull/51/commits/da4e6635eb43c709a5db812c94bb16003242d969.

File: src/IK/include/BiomechanicalAnalysis/IK/InverseKinematics.h [hunk 6]
```diff
@@ -391,7 +391,7 @@ class HumanIK
      * @param verticalForce vertical force
      * @return true if the orientation setpoint is set correctly
      */
-    bool updateFloorContactTask(const int node, const double verticalForce);
+    bool updateFloorContactTask(const int node, const double verticalForce, const double linkHeight);
```
[the remark this answers is not in the document]
Comment (evelyd) [reply src/IK/include/BiomechanicalAnalysis/IK/InverseKinematics.h]: Resolved in https://github.com/ami-iit/biomechanical-analysis-framework/pull/51/commits/da4e6635eb43c709a5db812c94bb16003242d969.

**Review by davidegorbani — APPROVED** [review]
Thank you! It looks good to me!

**Pull request merged**